### PR TITLE
Bunch of cleanups for verbs/RDM

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
Per commits:
1) CQ opening difference for RDM/MSG replaced by domain->rdm statement instead of attr->format
2) Rename request matching functions
3) Remove FI_IBV_ERROR macro, used VERBS_INFO. Add return code in case of unsuccessful repost receives
4) Error handling in fi_ibv_open_rdm_ep
5) Fix 6 coverity issues
6) verbs: verbs_util macros do not use fi_log correctly #1463

### CODE REVIEW

File: prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c [hunk 1]
```diff
@@ -50,7 +50,7 @@ int fi_ibv_rdm_start_connection(struct fi_ibv_rdm_ep *ep,
 		return 0;
 
 	if (ep->is_closing) {
-		FI_IBV_ERROR("Attempt to start connection with addr "
+		VERBS_INFO(FI_LOG_CORE, "Attempt to start connection with addr "
```
Comment (shefty) [inline prov/verbs/src/ep_rdm/verbs_av_ep_rdm.c]: Better option is FI_LOG_AV

File: prov/verbs/src/ep_rdm/verbs_ep_rdm.c [hunk 2]
```diff
@@ -389,20 +392,20 @@ static inline int fi_ibv_rdm_tagged_test_addr(const char *devname,
 	struct rdma_cm_id *test_id;
 	int test = 0;
 	if (rdma_create_id(NULL, &test_id, NULL, RDMA_PS_TCP)) {
-		FI_IBV_ERROR("Failed to create test rdma cm id: %s\n",
+		VERBS_INFO(FI_LOG_CORE, "Failed to create test rdma cm id: %s\n",
```
Comment (shefty) [inline prov/verbs/src/ep_rdm/verbs_ep_rdm.c]: Is the log selection here and below better as FI_LOG_CORE or FI_LOG_AV or FI_LOG_EP_CTRL?

File: prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm.c [hunk 3]
```diff
@@ -413,16 +410,18 @@ fi_ibv_rdm_tagged_process_recv(struct fi_ibv_rdm_ep *ep,
...
-				     tagmask);
-				abort();
+				VERBS_INFO(FI_LOG_EP_DATA,
+					"%s: %d RECV TRUNCATE, data_len=%d, posted_len=%d, "
+					"conn %p, tag 0x%llx, tagmask %llx\n",
+					__FUNCTION__, __LINE__, data_len,
+					found_request->len, found_request->conn,
+					found_request->tag,
+					found_request->tagmask);
+//				return -FI_EOTHER;
```
Comment (shefty) [inline prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm.c]: Note that the return statement is commented out.  I'm not sure if this is intentional.  Would an assert() be useful here?

**Comment by shefty** [comment]
Thanks - I just had a couple of comments.  Please either update the affected patches or just respond indicating that the changes you have are correct.

**Comment by evgeny-leksikov** [comment]
Thanks! Updated. BTW, is there generic possibility to diverse debug logs from large subsystems? cat/grep helps in some cases but usually very detailed logging changes behavior.

**Comment by shefty** [comment]
I'm not sure I'm following your comment.  We can enable/disable the logging per provider, per subsystem, with different debug levels assigned to each message.  Are you wanting more subsystems or more debug levels?

**Comment by evgeny-leksikov** [comment]
I'm wanting more debug levels which are absolutely excluded from release build like FI_DBG. FI_LOG adds extra if-statement on critical path, so I can't use it here for splitting by levels.

**Pull request merged**